repo/fsrepo/migrations/ipfsdir.go

```diff
@@ -0,0 +1,129 @@
+package migrations
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
+)
+
+const (
+	envIpfsPath = "IPFS_PATH"
+	defIpfsDir  = ".ipfs"
+	versionFile = "version"
+
+	// Local IPFS API
+	apiFile = "api"
+)
+
+func init() {
+	homedir.DisableCache = true
+}
+
+// ApiEndpoint reads the api file from the local ipfs install directory and
+// returns the address:port read from the file.  If the ipfs directory is not
+// specified then the default location is used.
+func ApiEndpoint(ipfsDir string) (string, error) {
+	ipfsDir, err := CheckIpfsDir(ipfsDir)
+	if err != nil {
+		return "", err
+	}
+	apiPath := path.Join(ipfsDir, apiFile)
+
+	apiData, err := ioutil.ReadFile(apiPath)
+	if err != nil {
+		return "", err
+	}
+
+	val := strings.TrimSpace(string(apiData))
+	parts := strings.Split(val, "/")
+	if len(parts) != 5 {
+		return "", fmt.Errorf("incorrectly formatted api string: %q", val)
+	}
+
+	return parts[2] + ":" + parts[4], nil
```

Comment:
This code is currently in ipfs-update. It looks like it's probably fine although it seems a bit brittle. @Stebalien do we only really support `/ipX/address/tcp/port` in the API file, or at least only support that for direct HTTP usage?

Perhaps switching over to go-ipfs-http-client would put us in a better spot here https://github.com/ipfs/go-ipfs-http-client/blob/3af36210f80fb86aae50da582b494ceddd64c3de/api.go#L99.

----

@gammazero if we do then it's reasonable to push the switch from go-ipfs-api to go-ipfs-http-client into a separate PR since it's unrelated to the migrations work